### ISSUE TITLE
s3TransferManager is uploading files with Content-Encoding metadata empty

### ISSUE THREAD

**Issue by swilsonmelo** [aws/aws-sdk-java-v2#4949]
### Describe the bug

s3TransferManager is uploading files with `Content-Encoding` metadata empty, we are uploading some files to s3 with the crt Async client, and the metadata of the objects in s3 contains the Content-Encoding key with an empty value.

![image](https://github.com/aws/aws-sdk-java-v2/assets/34746520/20598ed6-416d-4e87-9aaf-308cab1858cb)

We are uploading text files and we do not need this `Content-Enconding`

We tried setting it to null from the code, but it was still created with an empty value.

### Expected Behavior

Objects in s3 without the key Content-Encoding in metadata

### Current Behavior

Objects in s3 with the key Content-Encoding in metadata
![image](https://github.com/aws/aws-sdk-java-v2/assets/34746520/08b4d38a-e0f4-47cc-a012-b2ff4a631660)

### Reproduction Steps

Dependencies used in the project:

```
    implementation(platform("software.amazon.awssdk:bom:2.20.42"))
    implementation("software.amazon.awssdk:s3:2.20.42")
    implementation("software.amazon.awssdk:s3-transfer-manager:2.20.42")
    implementation("software.amazon.awssdk.crt:aws-crt:0.29.10"
```

Code where the s3Client and the s3TransferManager are being instantiated:
```kotlin
    @Bean
    fun S3AsyncClient(s3Properties: S3Properties): S3AsyncClient =
        S3AsyncClient.crtBuilder()
            .region(Region.of(s3Properties.region))
            .targetThroughputInGbps(s3Properties.targetThroughputInGbps)
            .minimumPartSizeInBytes(s3Properties.minimumPartSizeInMegaBytes * MB)
            .build()

    @Bean
    fun S3TransferManager(s3AsyncClient: S3AsyncClient): S3TransferManager =
        S3TransferManager.builder()
            .s3Client(s3AsyncClient)
            .build()
```

Code that we are using to upload a directory to s3:

```kotlin
override fun uploadDirectoryAsync(destinationFolder: String, directory: Path): CompletableFuture<URI> {
        val directoryUpload = s3TransferManager.uploadDirectory(
            UploadDirectoryRequest.builder()
                .source(directory)
                .bucket(s3Properties.recordingsBucketName)
                .s3Prefix(destinationFolder)
                .build()

        )
        val directoryUriParameters = mapOf("bucket" to s3Properties.recordingsBucketName, "path" to destinationFolder)
        return directoryUpload.completionFuture().thenApply { uriTemplate.expand(directoryUriParameters) }
    }
```
   

### AWS Java SDK version used

implementation(platform("software.amazon.awssdk:bom:2.20.42")) implementation("software.amazon.awssdk:s3:2.20.42") implementation("software.amazon.awssdk:s3-transfer-manager:2.20.42") implementation("software.amazon.awssdk.crt:aws-crt:0.29.10")

### JDK version used

java 21

### Operating System and version

Amazon linux 2 and running locally in MAC OS Sonoma 14.2.1

**Comment by debora-ito** [comment]
@swilsonmelo some questions to help with the troubleshooting:
- What's the size of the file? The upload is single or multipart?
- Can you generate the CRT [verbose logs](https://docs.aws.amazon.com/sdk-for-java/latest/developer-guide/logging-slf4j.html#sdk-java-logging-verbose)? Make sure to redact any sensitive information (like access keys)
- Can you test using 'software.amazon.awssdk.crt:aws-crt' 0.21.7 (not 0.29.10)? It's the expected version for SDK 2.20.42 ([link](https://github.com/aws/aws-sdk-java-v2/blob/fe3d2ef817998d1c622d8e670cbcd12178ba2370/pom.xml#L117C25-L117C31)), we recommend always using the same version the SDK expects. You can also test with a more recent SDK version like 2.24.0, which uses 'aws-crt' 0.29.9

**Comment by debora-ito** [comment]
@swilsonmelo after some research I found out this issue was reported before in the Python SDK. As per specification, the SDK sends `Content-Encoding: aws-chunked` but S3 is persisting the metadata with an empty value. I don't think S3 should persist this header at all.

This is an issue on the S3 side, we raised an internal ticket with them. We are also tracking progress in the cross-SDK repo here: https://github.com/aws/aws-sdk/issues/498, so I'll close this one. We'll post any updates in the cross-SDK issue.
